src/custom/state/orders/updaters/PendingOrdersUpdater.ts

```diff
@@ -33,42 +58,58 @@ export function PendingOrdersUpdater(): null {
         return
       }
 
-      // Iterate over pending orders fetching operator order data, async
+      // Iterate over pending orders fetching API data
       const unfilteredOrdersData = await Promise.all(
         pendingRef.current.map(async (orderFromStore) => fetchOrderPopupData(orderFromStore, chainId))
       )
 
       // Group resolved promises by status
       // Only pick the status that are final
-      const { fulfilled, expired, cancelled } = unfilteredOrdersData.reduce<
-        Record<ApiOrderStatus, OrderLogPopupMixData[]>
+      const { fulfilled, expired, cancelled, presigned } = unfilteredOrdersData.reduce<
+        Record<OrderTransitionStatus, OrderLogPopupMixData[]>
       >(
         (acc, { status, popupData }) => {
           popupData && acc[status].push(popupData)
           return acc
         },
-        { fulfilled: [], expired: [], cancelled: [], unknown: [], pending: [] }
+        { fulfilled: [], expired: [], cancelled: [], unknown: [], presigned: [], pending: [] }
       )
 
-      // Bach state update per group, if any
+      if (presigned.length > 0) {
+        // Only mark as presigned the orders we were not aware of their new state
+        const presignedOrderIds = presigned as OrderID[]
+        const ordersPresignaturePendingSigned = _getPresignedOrdersStillWaiting(pendingRef.current, presignedOrderIds)
+
+        if (ordersPresignaturePendingSigned.length > 0) {
+          markOrdersAsPresigned({
+            ids: ordersPresignaturePendingSigned,
+            chainId,
```

Comment:
Actually, do you need this at all?

If all you need are the ids, why not use `presignedOrderIds` ?

Reply:
Cause, I want to call the action only once. The classify orders only uses API data, it has not knowledge about our internal state. With this, i just get the subset of presigned ids that we weren't aware that are signed

Reply:
I still don't understand why you need to do that :thinking:

Reply:
I want to invoke `markOrdersAsPresigned` only for the **first time i see one pre-signed order has been signed**.

If I don't conditionally trigger the action, it will trigger `markOrdersAsPresigned` on every polling we do when they are signed and not executed. 

Happy to explain personally if it helps.

Reply:
I think I got it now, although not 100% agree.

My initial thinking was that the function was unnecessary because `preSignedOrders` is a subset of `pendingOrders`, so I did not get why you'd need to filter them out.

I see that is because you classify orders that have been presigned with `presigned` state, instead of open. So you just want to do this transition once.

What I don't agree is to map orders that were presigned with a status other than `open`.
I my view, the difference is in the pending state. Once it's pre-signed, all orders are equal. Well, sort of.

Reply:
Not sure if I understand 100% what you mean. I do agree, presign orders are in the same conceptual state as open. 

> I see that is because you classify orders that have been presigned with presigned state, instead of open

However, I understand with this you refer to this classification logic:

https://github.com/gnosis/cowswap/blob/9c3d6d923b0916948dee9d19978b7ff0a2302876/src/custom/state/orders/utils.ts#L53-L53

This logic is used in the updater for pending orders. This updater, hands the classifier the information return from the API for the orders that the apps thinks are pending. This why, it classifies as `presign` something that was "pending", meaning, we need to inform our state that something that is "pre-sign pending" needs to be marked as presigned.

When we invoke the `preSignOrders` action, we set the PENDING state, that is indeed what I understand you want --> set the state as OPEN.

>What I don't agree is to map orders that were presigned with a status other than open.

This is where i don't understand you, cause I don't map to any other state than OPEN (well, we call it PENDING), the `presign` classification, is just to decide based on API data which redux action we need to take depending on the new information.

Reply:
I guess I'm over-complicating things while not fully understanding what you are doing.

Go on, I'll comment again if i have another question